src/block/tile/Sign.php

```diff
@@ -78,7 +78,10 @@ public function __construct(World $world, Vector3 $pos){
 	}
 
 	public function readSaveData(CompoundTag $nbt) : void{
-		if(($textBlobTag = $nbt->getTag(self::TAG_TEXT_BLOB)) instanceof StringTag){ //MCPE 1.2 save format
+		$frontTextTag = $nbt->getTag(self::TAG_FRONT_TEXT);
+		if($frontTextTag instanceof CompoundTag){
+			$this->readSaveData($frontTextTag);
```

Comment:
I don't like this recursion. A separate helper method to read the text properties would be more appropriate. We'll need such a helper to handle back text once that's implemented too.